src/adams_methods.jl

```diff
@@ -0,0 +1,655 @@
+                # Else, then step size is rejected, reattempt with fac*next_h
+                if err_kp1 <= 1
+                    successful_step = true
+                else
+                    successful_step = false
+                    h = fac*next_h # make sure fac < 1
+                end
+            end # End of error check for current step size h
+        end # End of loop for finding successfull step size for nth step
+            # repeats if success_step == false
+
+
+
+        #--#####################################################################
+        # Interpolation for dense output
+        #--#####################################################################
+        # interpolate specified points
+        while iter <= length(tspan) && tdir*tspan[iter]<tdir*t_np1
+            h_iter = tspan[iter]-t[n]
+
+            # Calculation of g_{j}(n)coefficients for j=0,...,k+1;
+            g_coefs!(g,c,h,t,tspan[iter],n,min(n,steporder+1))
+
+            # Calculation of  ϕstar_{j}(n), and ϕ_{j}(n) coef. for j=0,...,k;
+            ϕ_and_ϕstar_coefs!(b, ϕ, ϕstar,dy[n],t,tspan[iter],n,1:min(max(n-1,1),steporder+1))
+
+            yout_iter = y[n]
+            for j= 1:steporder
+                yout_iter += h_iter*g[j]*ϕstar[2,j]
+            end
+
+            #CE iteration for more precise interpolation
+            if n>=steporder+1
+                dy_iter=F(tspan[iter],yout_iter)
+                ϕ_coefs!(ϕ, ϕstar,dy_iter,1:steporder+1)
+                yout_iter = yout_iter + h_iter*g[steporder+1]*ϕ[3,steporder+1]
+            end
+
+            push!(tout,tspan[iter])
+            push!(yout, yout_iter)
+            iter += 1
+        end
+
+        if iter <= length(tspan) && tspan[iter] == t_np1
+            # handling when tspan and choose time steps overlap
+            push!(tout, t_np1)
+            push!(yout, y_np1)
+            iter += 1
+        elseif points == :all
+            push!(tout, t_np1)
+            push!(yout, y_np1)
+        end
+
+        # restore values of coefficients for choosen step size
+        g_coefs!(g,c,h,t,t_np1,n,min(n,steporder+1))
+        ϕ_and_ϕstar_coefs!(b, ϕ, ϕstar,dy[n],t,t_np1,n,1:min(max(n-1,1),steporder+1))
+        ϕ_coefs!(ϕ, ϕstar,dy_np1,1:steporder+1)
+
+        # push work variables
+        push!(y,y_np1)
+        push!(dy,dy_np1)
+        push!(t,t_np1)
+
+        #--#####################################################################
+        # Variable order selection for next step
+        #   Note run until a success step was taken for n-th step
+        #   See Harier pg 424 for indepeth description
+        #--#####################################################################
+
+        current_steporder = steporder
+        if n<=order+1
+            # Need to run the first few steps at reduced order for it requires
+            # the last 'order' number of points
+            steporder = n==1 ? 1 : n-1
+
+        elseif adaptive_order && steporder>1
```

Comment:
Notice in line 482, we need the difference 
```
g[steporder] - g[steporder-1]
```
and so if `steporder` is one, then `g[steporder-1]` is out of range.